packages/web/src/components/gcds-select/gcds-select.tsx

```diff
@@ -274,6 +279,22 @@ export class GcdsSelect {
     }
 
     this.inheritedAttributes = inheritAttributes(this.el, this.shadowElement);
+
+    if (this.el.children) {
+      this.options = Array.from(this.el.children);
+      for (let x = 0; x < this.options.length; x++) {
+        if (this.options[x].hasAttribute('selected')) {
+          this.value = this.options[x].getAttribute('value');
+          this.initialValue = this.value ? this.value : null;
+        }
+
+        if (this.value == this.options[x].getAttribute('value')) {
+          this.options[x].setAttribute('selected', 'true');
+          this.internals.setFormValue(this.options[x].getAttribute('value'));
+          this.initialValue = this.value;
+        }
+      }
```

Comment:
Could we use map instead?
```suggestion
      this.options.map((option) => {
        if (option.hasAttribute('selected')) {
          this.value = option.getAttribute('value');
          this.initialValue = this.value ? this.value : null;
        }

        if (this.value === option.getAttribute('value')) {
          option.setAttribute('selected', 'true');
          this.internals.setFormValue(option.getAttribute('value'));
          this.initialValue = this.value;
        }
      });
```